src/transactions/InflationOpFrame.cpp

```diff
@@ -108,7 +108,7 @@ InflationOpFrame::doApply(Application& app, LedgerDelta& delta,
             {
                 lcl.totalCoins += toDoleThisWinner;
             }
-            if (!winner->addBalance(toDoleThisWinner))
+            if (!winner->addBalance(toDoleThisWinner, ledgerManager))
             {
                 throw std::runtime_error(
```

Comment:
This throw was added here as it was impossible for above condition even to get true.

But is it impossible now? Imagine an issuer that creates asset that requires authorization. It sends INT64_MAX of that asset to one account that has enough XLM to be a winner in inflation. Then that account can make an offer to sell this asset 1:1 for XLM, making its buying liabilities so big, that its addBalance returns 0. Then at the next inflation round all nodes will drop this transaction with txINTERNAL_ERROR, making inflation operations no-op for as long as they want.

I think we need another solution for handling that case now.

Reply:
Good catch @vogel : I think you are correct, `toDoleThisWinner` should be capped by the amount that the account can receive instead of just being `bigDivide(amountToDole, w.mVotes, totalVotes, ROUND_DOWN)`.

Reply:
Good catch indeed. Fixed as described above.